staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -184,8 +184,11 @@ func (s *EtcdOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.DurationVar(&s.StorageConfig.HealthcheckTimeout, "etcd-healthcheck-timeout", s.StorageConfig.HealthcheckTimeout,
 		"The timeout to use when checking etcd health.")
 
-	fs.Int64Var(&s.StorageConfig.LeaseReuseDurationSeconds, "lease-reuse-duration-seconds", s.StorageConfig.LeaseReuseDurationSeconds,
+	fs.Int64Var(&s.StorageConfig.LeaseManagerConfig.ReuseDurationSeconds, "lease-reuse-duration-seconds", s.StorageConfig.LeaseManagerConfig.ReuseDurationSeconds,
 		"The time in seconds that each lease is reused. A lower value could avoid large number of objects reusing the same lease. Notice that a too small value may cause performance problems at storage layer.")
+
+	fs.Int64Var(&s.StorageConfig.LeaseManagerConfig.MaxObjectSize, "lease-max-object-size", s.StorageConfig.LeaseManagerConfig.MaxObjectSize,
+		"The max object size that each lease can attach. This option is under tuning, the default value is not a recommended value, just a placeholder currently. A lower value could avoid large number of objects reusing the same lease. Notice that a too small value may cause performance problems at storage layer.")
```

Comment:
Once we make a flag, we shouldn't change the default if possible. I think we will can come up with a reasonable default at this point.

Reply:
I never tested etcd lease performance before. What's your recommended value here?

Reply:
I'm think maybe 5k? @mborsz - WDYT?

Reply:
no idea :D

I suggest splitting this PR into two:
* no behavioral change, only metrics + log below to understand the current values (it should be easily accessible in our 5k tests)
* actual implementing lease limiting with value based on the above

Reply:
Done. Once this get merged and we found a reasonable default value, update #96836 and I will submit a PR with `--lease-max-object-count`